flytekit_scripts/flytekit_build_image.sh

```diff
@@ -61,10 +61,17 @@ if [ -n "$REGISTRY" ]; then
 else
   FLYTE_INTERNAL_IMAGE=${IMAGE_NAME}:${PREFIX}${TAG}
 fi
-echo "Building: $FLYTE_INTERNAL_IMAGE"
+
+# Check if the user set the target build architecture, if not use the default instead.
+if [ -z "$TARGET_PLATFORM_BUILD" ]; then
```

Comment:
Maybe we can simplify this by skipping `--platform` so Docker will do things accordingly:
* If there is a base image matching the current architecture, it will build an image matching the architecture
* If there is no base image matching the current architecture, it will do a build matching the base image architecture

I think if we force `--platform` and there is base image matching that, it will fail. If we think it is better to fail loudly, then it is fine to pass `--platform` explicitly, but this will be a breaking change that we need to note.

Reply:
But I think that this doesn't solve the main problem. The problem is that you want to build the new image to run in a different architecture than where you are running this image.

>If there is a base image matching the current architecture, it will build an image matching the architecture

If we go into this case, it is possible that you want to run the image in a different architecture. Is your solution to this only having the image base available in the running architecture?

Reply:
ok I think that I understood your proposal, I'm going to change it :)

Reply:
👍🏼 done!